src/main/java/org/sagebionetworks/bridge/adherence/WeeklyAdherenceReportWorkerProcessor.java

```diff
@@ -0,0 +1,162 @@
+            LOG.info("Weekly adherence report caching took " + requestStopwatch.elapsed(TimeUnit.SECONDS) + " seconds");
+        }
+    }
+    
+    private void process(WeeklyAdherenceReportRequest request) throws Exception {
+        ForWorkersApi workersApi = clientManager.getClient(ForWorkersApi.class);
+        
+        List<App> appList = null;
+        if (request.getAppId() != null) {
+            appList = ImmutableList.of( bridgeHelper.getApp(request.getAppId()) );
+        } else {
+            appList = bridgeHelper.getAllApps();    
+        }
+        for (App app : appList) {
+            Map<String, Integer> studyThresholds = new HashMap<>();
+            
+            // First go through the studies and select the studies which should have reports generated, and note their
+            // threshold criteria if it is set 
+            PagedResourceIterator<Study> studyIterator = new PagedResourceIterator<>((ob, ps) -> 
+                workersApi.getAppStudies(app.getIdentifier(), ob, ps, false).execute().body().getItems(), PAGE_SIZE);
+            
+            while(studyIterator.hasNext()) {
+                Study study = studyIterator.next();
+                // This excludes legacy studies, and thus excludes all older apps, and well as any studies
+                // that have no schedules. Right now this is hundreds of studies with thousands of users, but
+                // we will need to continue to define limits to this processing as the system grows.
+                if (ACTIVE_PHASES.contains(study.getPhase()) && study.getScheduleGuid() != null) {
+                    studyThresholds.put(study.getIdentifier(), getThresholdPercentage(study));
+                }
+            }
+            if (studyThresholds.isEmpty()) {
+                LOG.info("Skipping app " + app.getIdentifier() + ": it has no reportable studies");
+                continue;
+            } else {
+                LOG.info("Caching studies in app " + app.getIdentifier() + " starting at " + DateTime.now());
+            }
+            Stopwatch appStopwatch = Stopwatch.createStarted();
+            
+            // Now go through all accounts that are enrolled in at least one study and push cache a report for each study.
+            PagedResourceIterator<AccountSummary> acctIterator = new PagedResourceIterator<>((ob, ps) -> {
+                AccountSummarySearch search = new AccountSummarySearch().offsetBy(ob).pageSize(ps).enrollment(ENROLLED);
+                return workersApi.searchAccountSummariesForApp(app.getIdentifier(), search).execute().body().getItems();
+            }, PAGE_SIZE);
+            
+            while (acctIterator.hasNext()) {
+                AccountSummary summary = acctIterator.next();
+                for (String studyId : summary.getStudyIds()) {
+                    if (request.getStudyId() != null && !studyId.equals(request.getStudyId())) {
+                        continue;
+                    }
```

Comment:
We should add a short unit test to test that this filters out studies correctly.

Reply:
Added